external/boost/archive/portable_binary_archive.hpp

```diff
@@ -37,7 +37,8 @@ enum portable_binary_archive_flags {
 //#endif
 
 inline void
-reverse_bytes(char size, char *address){
+reverse_bytes(signed char size, char *address){
+    BOOST_ASSERT(size > 0);

```

Comment:
I'd use >=, maybe. reverse_bytes sounds like something only used for endianness, so may not matter in practice.
Also an throw is probably best, since it doesn't end the program, but fails the serialization.